src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -449,5 +454,10 @@ internal static ClassType GetClassType(Type type)
 
             return ClassType.Object;
         }
+
+        internal static bool IsNonGenericInterface(Type type)
```

Comment:
Nit: IsSupportedNonGenericInterface, or something like that?  There are plenty of non-generic interface types that this will return false for.

Reply:
Based on https://github.com/dotnet/corefx/pull/38307#discussion_r291748224, I've removed this method and updated its dependencies to revolve around a more general `typeof(IList).IsAssignableFrom(type)` check.